hw02-generics/src/main/java/ru/otus/hw02/DIYarrayList.java

```diff
@@ -0,0 +1,214 @@
+package ru.otus.hw02;
+
+
+import java.util.*;
+
+public class DIYarrayList<T> implements List<T> {
+    private Object[] array;
+    private int capacity = 0;
+    private final int INIT_SIZE = 10;
+
+    public DIYarrayList() {
+        this.array = new Object[INIT_SIZE];
+    }
+
+    public DIYarrayList(int setSize) {
+        this.array = new Object[setSize];
+    }
+
+    public DIYarrayList (List<T> list) {
+        if ((capacity = list.size()) != 0)
```

Comment:
Таких конструкций лучше избегать. Не все привычны к такому, Это на собеседованиях хорошо подобное спрашивать.
Лучше разбить на несколько строк.
```suggestion
        capacity = list.size();
        if (capacity != 0)
```